components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -153,14 +218,7 @@ fn path_list(args: &[ScalarValueRef]) -> Result<Option<Vec<PathExpression>>> {
     for arg in args {
         let json_path: &Option<Bytes> = arg.as_ref();
 
-        let json_path = match json_path.as_ref() {
-            None => return Ok(None),
-            Some(p) => std::str::from_utf8(&p).map_err(crate::codec::Error::from),
-        }?;
-
-        let path_expr = parse_json_path_expr(&json_path)?;
-
-        path_expr_list.push(path_expr);
+        path_expr_list.push(try_opt!(encode_to_json_path(json_path)));
```

Comment:
so why do we need a function called `encode_to_json_path` instead of directly write the logic here?

Reply:
> so why do we need a function called `encode_to_json_path` instead of directly write the logic here?

To avoid repeating using 
```rust 
let path = match ... 
let path_expr = parse_json_path_expr(&path)?;
```

Reply:
So looks like `encode_to_json_path` should be called `parse_json_path` and `path_list` should be called `parse_json_path_list`?

Reply:
> So looks like `encode_to_json_path` should be called `parse_json_path` and `path_list` should be called `parse_json_path_list`?

I think I'm not good in naming XD, that's really a good idea